### PR TITLE
LPD-35168 Binding two trees into one using a relationship

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPD-35168](https://liferay.atlassian.net/browse/LPD-35168)

[LPD-35168]: https://liferay.atlassian.net/browse/LPD-35168?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by pedro-oliveira446** [comment]
ci:test:sf

File: modules/apps/object/object-api/src/main/java/com/liferay/object/tree/TreeFactory.java [hunk 1]
```diff
@@ -6,21 +6,168 @@
...
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
 
 /**
  * @author Feliphe Marinho
  */
-public interface TreeFactory {
+public class TreeFactory {
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/tree/TreeFactory.java:29]: Remove this class, it is no longer needed

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
Sent [here](https://github.com/liferay-core-infra/liferay-portal/pull/7496) too

**Comment by pedro-oliveira446** [comment]
ci:report:31418871

**Pull request closed without merging**